crates/miden-objects/src/address/mod.rs

```diff
@@ -0,0 +1,83 @@
+use crate::AddressError;
+use crate::account::{AccountId, AccountStorageMode};
+use crate::note::NoteTag;
+
+/// A user-facing address in Miden.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum Address {
+    AccountId(AccountIdAddress),
+}
+
+/// Address that targets a specific `AccountId` with an explicit tag length preference.
+///
+/// The tag length preference lets the owner of the account choose their level of privacy. A higher
+/// tag length makes the account more uniquely identifiable and reduces privacy, while a shorter
+/// length increases privacy at the cost of matching more notes published onchain.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct AccountIdAddress {
+    id: AccountId,
+    tag_len: u8,
+}
+
+impl AccountIdAddress {
+    /// Creates a new account-id based address with the default tag length.
```

Comment:
```suggestion
    /// Creates a new account ID based address with the default tag length.
```

Nit: At some point we decided to use "account ID" spelling and we should already use it mostly everywhere.